File: 2018/day04/day04_test.go
package main

import (
	"testing"
)

func TestPart1(t *testing.T) {
	lines := readFile("day04.test.txt")

	expected := 240
	actual := part1(lines)
	if expected != actual {
		t.Fatalf("Expected %v but was %v", expected, actual)
	}
}

func TestPart2(t *testing.T) {
	lines := readFile("day04.test.txt")

	expected := 4455
	actual := part2(lines)
	if expected != actual {
		t.Fatalf("Expected %v but was %v", expected, actual)
	}
}

File: solutions/2025/day04/tests/day04_test.py
from day04.day04 import Day04
import unittest


class TestDay04(unittest.TestCase):
    def test_part1(self):
        input = Day04().read_file("day04/tests/day04.test.txt")
        self.assertEqual(Day04.solve_part1(input), 13)

    def test_part2(self):
        input = Day04().read_file("day04/tests/day04.test.txt")
        self.assertEqual(Day04.solve_part2(input), 43)


if __name__ == "__main__":
    unittest.main()
